src/sankeyDiagram.ts

```diff
@@ -2038,98 +2036,116 @@ export class SankeyDiagram implements IVisual {
+        curveRadius = fixedLinkHeight + (SankeyDiagram.NodeAndBackwardLinkDistance + link.dyDestination)/2;
+        curveCenterY = link.destination.y - SankeyDiagram.NodeAndBackwardLinkDistance - fixedLinkHeight + curveRadius;
+        linkInnerKneeSize = (link.destination.selfLinkWeight? Math.min(link.destination.width, minHeight) + SankeyDiagram.DistanceBetweenLinks : 0) + SankeyDiagram.NodeAndBackwardLinkDistance;
+        fixedLinkKneeSize = linkKneeSize + SankeyDiagram.NodeAndBackwardLinkDistance;
+
+        if (fixedLinkKneeSize - linkInnerKneeSize < SankeyDiagram.DefaultNodeWidth){
+            fixedLinkKneeSize = SankeyDiagram.DefaultNodeWidth + linkInnerKneeSize;
         }
-
+        
         pathParams +=
-            `C ${link.destination.x + link.destination.width + linkKneeSize} ${link.destination.y + fixedLinkHeight - linkKneeSize},` +
-            ` ${link.destination.x + link.destination.width + linkKneeSize} ${link.destination.y + fixedLinkHeight} ,` +
-            ` ${link.destination.x + link.destination.width + distanceFromNodeToLinks} ${link.destination.y + fixedLinkHeight}`;
+            `C ${curveCenterX - link.destination.width/2} ${curveCenterY - curveRadius}, ` +
+            `${curveCenterX + fixedLinkKneeSize} ${curveCenterY - curveRadius}, ` +
+            `${curveCenterX + fixedLinkKneeSize} ${curveCenterY}`; 
 
-        pathParams += `L ${link.destination.x + link.destination.width} ${link.destination.y + fixedLinkHeight}`;
+        pathParams +=
+            `C ${curveCenterX + fixedLinkKneeSize} ${curveCenterY},` +
+            ` ${curveCenterX + fixedLinkKneeSize} ${link.destination.y + fixedLinkHeight + link.dyDestination + SankeyDiagram.DistanceBetweenLinks/2} ,` +
+            ` ${curveCenterX} ${link.destination.y + fixedLinkHeight + link.dyDestination}`;
 
-        pathParams += `L ${link.destination.x + link.destination.width} ${link.destination.y}`;
+        pathParams += `L ${curveCenterX} ${link.destination.y + link.dyDestination + SankeyDiagram.DistanceBetweenLinks/2}`;
 
-        pathParams += `L ${link.destination.x + link.destination.width + distanceFromNodeToLinks} ${link.destination.y}`;
+        curveRadius = (link.dyDestination + SankeyDiagram.NodeAndBackwardLinkDistance)/2; 
+        curveCenterY = link.destination.y - SankeyDiagram.NodeAndBackwardLinkDistance + curveRadius;
+       
+        pathParams +=
+            `C ${curveCenterX - link.destination.width/2} ${link.destination.y + link.dyDestination + SankeyDiagram.DistanceBetweenLinks/2},` +
+            ` ${curveCenterX + linkInnerKneeSize} ${link.destination.y + link.dyDestination + SankeyDiagram.DistanceBetweenLinks/2} ,` +
+            ` ${curveCenterX + linkInnerKneeSize} ${curveCenterY}`;      
 
-        pathParams += `L ${link.destination.x + link.destination.width + distanceFromNodeToLinks} ${link.destination.y - SankeyDiagram.NodeAndBackwardLinkDistance}`;
+        pathParams +=
```

Comment:
Is _**pathParams**_ variable's several additions can be separated to other function or calculated at the end?